src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,49 @@
+package duke;
+
+import tasks.TaskList;
+
+public class Parser {
+    private TaskList taskList;
+    public Parser(TaskList tList) {
+        taskList = tList;
+    };
+
+    /**
+     * Parses the input from user and checks if there are any invalid commands or format.
+     *
+     * @param str Input from User.
+     * @return String of the original user input if no violations are found.
+     * @throws MissingDescriptionException If a command is missing required information.
+     * @throws DukeException If the command or format is invalid.
+     */
+    public String parse(String str) throws MissingDescriptionException, DukeException, InvalidTaskException {
+        String[] strArr = str.split(" ");
+        String firstWord = strArr[0];
+        if (firstWord.equals("deadline") || firstWord.equals("event") || firstWord.equals("todo")
```

Comment:
Perhaps you could also consider placing these possible 'first words' into a private String[], and utilise the contains() method to check if the firstWord from the argument passed matches any of the array items? This might possibly help to avoid having to repeat similar lines of code in the conditional statement! :)

Reply:
Yes I didn't think of that. Thank you!